src/storage/src/statistics.rs

```diff
@@ -319,6 +353,48 @@ impl StorageStatistics<SourceStatisticsUpdate, SourceStatisticsMetrics> {
         cur.1.bytes_received = cur.1.bytes_received + value;
         cur.2.bytes_received.inc_by(value);
     }
+
+    /// Update the `envelope_state_bytes` stat.
+    /// A positive value will add and a negative value will subtract.
+    pub fn update_envelope_state_bytes_by(&self, value: i64) {
+        let mut cur = self.stats.borrow_mut();
+        cur.1.envelope_state_bytes = cur.1.envelope_state_bytes.saturating_add_signed(value);
+        let abs_value = value.unsigned_abs();
+        if value > 0 {
+            cur.2.envelope_state_bytes.add(abs_value);
+        } else {
+            cur.2.envelope_state_bytes.sub(abs_value);
+        }
```

Comment:
add supports negative numbers, so you dont need this: https://docs.rs/prometheus/latest/prometheus/core/struct.GenericGauge.html#method.add (but I would check if its 0 (and set it to 0 and emit a warning)

Reply:
The types weren't matching. The methods would only accept `u64` since `envelope_state_bytes` was u64

Reply:
Updated to add warnings, also making sure that the same value is set for both the table metric and the prometheus gauge (the behavior would have differed in case of overflows).